### PR TITLE
FE-874 - Add Matchbox Grid component wrapper

### PR DESCRIPTION
**Pull request by nicklemmon**
[FE-874](https://jira.int.messagesystems.com/browse/FE-874)

### What Changed
- Updates Hibana Matchbox to version `13`
- Adds component wrapper for Matchbox Grid component
- Adds supporting unit test

### How To Test
- Grids are everywhere! Click through the app to verify any jankiness
- A clear example is the `/reports/summary` page

### To Do
- [ ] Incorporate feedback

### CODE REVIEW

File: src/components/pieChart/Legend.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Grid } from '@sparkpost/matchbox';
+import { Grid } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/components/pieChart/Legend.js:3]: Only updated here - the rest is Prettier.

File: src/pages/billing/components/RecipientValidationModal.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import React from 'react';
-import { Grid } from '@sparkpost/matchbox';
-import { Panel } from 'src/components/matchbox';
+import { Grid, Panel } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/billing/components/RecipientValidationModal.js:2]: Only updated here - the rest is Prettier.

File: src/pages/billing/forms/fields/BillingContactForm.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { connect } from 'react-redux';
 import { Field, formValueSelector, change } from 'redux-form';
 import PropTypes from 'prop-types';
 import _ from 'lodash';
-import { Grid } from '@sparkpost/matchbox';
+import { Grid } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/billing/forms/fields/BillingContactForm.js:6]: Only updated here - the rest is Prettier.

File: src/pages/inboxPlacement/components/AuthenticationResults.js [hunk 4]
```diff
@@ -1,18 +1,18 @@
 import React from 'react';
 import PercentWheel from 'src/components/percentWheel';
-import { Grid } from '@sparkpost/matchbox';
+import { Grid } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/inboxPlacement/components/AuthenticationResults.js:3]: Only updated here - the rest is Prettier.

File: src/pages/inboxPlacement/components/InfoBlock.js [hunk 5]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
-import { Grid } from '@sparkpost/matchbox';
+import { Grid } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/inboxPlacement/components/InfoBlock.js:2]: Only updated here - the rest is Prettier.

File: src/pages/inboxPlacement/components/TestDetails.js [hunk 6]
```diff
@@ -1,7 +1,6 @@
 import React, { useState, useCallback } from 'react';
 import { format } from 'date-fns';
-import { Grid } from '@sparkpost/matchbox';
-import { Label, Panel, Select, TextField, ScreenReaderOnly } from 'src/components/matchbox';
+import { Grid, Label, Panel, Select, TextField, ScreenReaderOnly } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/inboxPlacement/components/TestDetails.js:3]: Only updated here - the rest is Prettier.

File: src/pages/inboxPlacement/components/TimeToReceiveSection.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { Grid } from '@sparkpost/matchbox';
+import { Grid } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/inboxPlacement/components/TimeToReceiveSection.js:2]: Only updated here - the rest is Prettier.

File: src/pages/reports/accepted/components/AcceptedChart.js [hunk 8]
```diff
@@ -1,7 +1,6 @@
 import React, { Component } from 'react';
 import _ from 'lodash';
-import { Grid } from '@sparkpost/matchbox';
-import { Panel } from 'src/components/matchbox';
+import { Grid, Panel } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/reports/accepted/components/AcceptedChart.js:3]: Only updated here - the rest is Prettier.

File: src/pages/reports/accepted/components/TopLevelMetrics.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { Grid } from '@sparkpost/matchbox';
+import { Grid } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/reports/accepted/components/TopLevelMetrics.js:2]: Only updated here - the rest is Prettier.

File: src/pages/reports/bounce/components/BounceChart.js [hunk 10]
```diff
@@ -1,7 +1,6 @@
 import React, { Component } from 'react';
 import _ from 'lodash';
-import { Grid } from '@sparkpost/matchbox';
-import { Panel } from 'src/components/matchbox';
+import { Grid, Panel } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/reports/bounce/components/BounceChart.js:3]: Only updated here - the rest is Prettier.

File: src/pages/reports/components/MetricsSummary.js [hunk 11]
```diff
@@ -1,25 +1,34 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { Grid } from '@sparkpost/matchbox';
-import { Panel } from 'src/components/matchbox';
+import { Grid, Panel } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/reports/components/MetricsSummary.js:4]: Only updated here - the rest is Prettier.

File: src/pages/signals/dashboards/EngagementRecencyDashboard.js [hunk 12]
```diff
@@ -1,8 +1,7 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { list as getSubaccounts } from 'src/actions/subaccounts';
-import { Grid } from '@sparkpost/matchbox';
-import { Panel } from 'src/components/matchbox';
+import { Grid, Panel } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/signals/dashboards/EngagementRecencyDashboard.js:4]: Only updated here - the rest is Prettier.

File: src/pages/signals/dashboards/SpamTrapDashboard.js [hunk 13]
```diff
@@ -1,8 +1,7 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { list as getSubaccounts } from 'src/actions/subaccounts';
-import { Grid } from '@sparkpost/matchbox';
-import { Panel } from 'src/components/matchbox';;
+import { Grid, Panel } from 'src/components/matchbox';
```
Comment (nicklemmon) [inline src/pages/signals/dashboards/SpamTrapDashboard.js:4]: Only updated here - the rest is Prettier.

File: src/components/matchbox/Grid.js [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+
+  if (!isHibanaEnabled) {
+    return <OGGrid.Column {...omitSystemProps(props)} />;
+  }
+
+  return <HibanaGrid.Column {...props} />;
+}
+
+Grid.Column = Column;
+HibanaGrid.Column.displayName = 'HibanaGrid.Column';
```
Comment (kchu1337) [inline src/components/matchbox/Grid.js:30]: Probably need something like ```Column.displayName = 'Grid.Column';``` to fix your snapshot tests

**Comment by kchu1337** [comment]
Also the Alerts cards are broken but the template ones are working (in Hibana style). Not sure why 🤷‍♂

File: src/components/matchbox/Grid.js (hunk 14 above)
Comment (nicklemmon) [reply src/components/matchbox/Grid.js:30]: Yep, definitely. Got it fixed.

**Comment by nicklemmon** [comment]
> Also the Alerts cards are broken but the template ones are working (in Hibana style). Not sure why 🤷‍♂

@kchu1337 it looks like the Alerts cards are using some custom CSS applied via `className`. Looks like in Hibana the `<Panel/>` doesn't expose `className` as a prop oddly enough - or it just isn't applied to the same DOM element.

**Pull request merged**